### PR TITLE
docs: update of readthedocs config and docs/source/conf.py

### PR DESCRIPTION
**Pull request by consideRatio**
Refactoring and chore updates to docs/conf.py and .readthedocs.yaml

## Related
- https://github.com/jupyterhub/binderhub/pull/1565
- https://github.com/jupyterhub/repo2docker/pull/1197

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -224,8 +126,32 @@ def parse_schema(d, md=[], depth=0, pre=""):
...
+html_favicon = "_static/images/logo/favicon.ico"
+html_static_path = ["_static"]
+html_css_files = ["custom.css"]
+
+# pydata_sphinx_theme reference: https://pydata-sphinx-theme.readthedocs.io/en/latest/
+html_theme = "pydata_sphinx_theme"
+html_theme_options = {
+    "github_url": "https://github.com/jupyterhub/zero-to-jupyterhub-k8s/",
+    "use_edit_page_button": True,
+    "twitter_url": "https://twitter.com/mybinderteam",
```
Comment (manics) [inline docs/source/conf.py:142]:
```suggestion
```

File: docs/source/conf.py [hunk 2]
```diff
@@ -72,114 +80,8 @@ def _get_git_ref_from_chartpress_based_version(version):
...
-project = "Zero to JupyterHub with Kubernetes"
-copyright = f"{datetime.date.today().year}, Project Jupyter Contributors"
-author = "Project Jupyter Contributors"
-
-
-# -- General Sphinx configuration ---------------------------------------------------
-# ref: https://www.sphinx-doc.org/en/master/usage/configuration.html#general-configuration
-
-# Set the default role so we can use `foo` instead of ``foo``
-default_role = "literal"
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply docs/source/conf.py:95]: Example

````
With default_role as literal in rST, you can write `test` to have it in code style
Without default_role as literal in rST, you must write ``test``to have it in code style
````

**Review by choldgraf — APPROVED** [review]
LGTM!

**Comment by consideRatio** [comment]
Thanks for review @choldgraf!! :heart: :tada: Going for a merge as I think this PR isn't justified taking up further review efforts

**Pull request merged**